src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+
+import org.apache.lucene.util.SetOnce;
+import org.opensearch.client.Client;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.io.stream.NamedWriteableRegistry;
+import org.opensearch.common.io.stream.NamedWriteableRegistry.Entry;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.settings.Setting.Property;
+import org.opensearch.common.xcontent.NamedXContentRegistry;
+import org.opensearch.env.Environment;
+import org.opensearch.env.NodeEnvironment;
+import org.opensearch.node.Node;
+import org.opensearch.repositories.RepositoriesService;
+import org.opensearch.script.ScriptContext;
+import org.opensearch.script.ScriptEngine;
+import org.opensearch.script.ScriptService;
+import org.opensearch.threadpool.ExecutorBuilder;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.watcher.ResourceWatcherService;
+
+public abstract class BaseExtension implements Extension {
+    private Client client;
+    private ClusterService clusterService;
+    private ThreadPool threadPool;
+    private ResourceWatcherService resourceWatcherService;
+    private ScriptService scriptService;
+    private NamedXContentRegistry xContentRegistry;
+    private Environment environment;
+    private NodeEnvironment nodeEnvironment;
+    private NamedWriteableRegistry namedWriteableRegistry;
+    private IndexNameExpressionResolver indexNameExpressionResolver;
+    private Supplier<RepositoriesService> repositoriesServiceSupplier;
+
+    /**
+     * 
+     */
+    public BaseExtension() throws IOException {
+        createComponents();
+    }
+
+    private void createComponents() throws IOException {
+        //Settings.EMPTY will eventually be replaced with a getSettings method from the Extension interface
+        Settings settings = Settings.builder().put("node.name", Property.NodeScope).build();
+        this.threadPool = new ThreadPool(settings, new AtomicReference<>(), Collections.emptyList().toArray(new ExecutorBuilder[0]));
+        this.client = new NodeClient(settings, threadPool);
```

Comment:
This may be the SDKClient from the `ExtensionsRunner`.  Another param to add to the arguments, seems @owaiskazi19 [comment here](https://github.com/opensearch-project/opensearch-sdk-java/issues/136#issuecomment-1251444056) was probably right despite my argument against. :)